Set the bright foreground color in gnome-dark

kepstin opened the pull request:
My original intention was that the gnome-dark scheme would use the colour corresponding to 'bright white' as the bright version of the foreground colour. (The default foreground colour is 'normal white'.) This didn't make it into the gnome-terminal code, because the colour scheme settings are not able to set the bright foreground colour. But it can be set here.

kepstin committed:
Set the bright foreground color in gnome-dark

My original intention was that the gnome-dark scheme would use the colour corresponding to 'bright white' as the bright version of the foreground colour. (The default foreground colour is 'normal white'.) This didn't make it into the gnome-terminal code, because the colour scheme settings are not able to set the bright foreground colour. But it can be set here.

kepstin commented:
Note that for best results, I recommend using the gnome-dark scheme in combination with setting `draw_bold_text_with_bright_colors: true`, is that something that should be included here?

leifmetcalf commented:
Thank you for your patch! This is a good idea, and we should do the same for the gnome-light theme as well, which I've done in a commit on your branch.

I don't think we should set `draw_bold_text_with_bright_colors` since a lot of people (including me, though with the light theme) don't use it.

leifmetcalf committed:
Set the bright foreground color in gnome-light

This is essentially the same commit as f05c0f325c50ba7424041d343511b80c1b7b1550 but for the light colour scheme.

kepstin commented:
For the gnome-light theme, I have no particular recommendations about `draw_bold_text_with_bright_colors` either way, the theme looks fine with both. It's just a problem on the dark theme specifically, since many applications (including the default `ls` colors on many distros) use bold blue and expect it to be readable - the normal blue is a bit too dark for that.

leifmetcalf commented:
I see. The contrast with `ls` and `gnome-dark` is pretty bad. I'll add a note in the gnome-dark scheme warning users of this problem and recommending setting `draw_bold_text_with_bright_colors`.
